src/WebJobs.Extensions.DurableTask/Options/AzureStorageAccountOptions.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Options
+{
+    internal sealed class AzureStorageAccountOptions
```

Comment:
Dumb question, but how do instances of this type get initialized and configured? I couldn't immediately tell by looking at the code. The other "options" types in this directory are set in host.json, but is that also true for AzureStorageAccountOptions? Or are we grabbing specific app settings to populate this? I see that you're pulling these objects from an `IConfigurationSection` that's injected by the host, but `IConfigurationSection` is unfortunately a bit of a black box for me.

Could we include some steps in the PR description that explain how to set up identity-based connection support?

Reply:
Not a dumb question at all. Without much first-hand use of `IConfiguration` and `IConfigurationSection`, it would be difficult to guess. A bit of background then!

The `IConfiguration` class is really a glorified `IDictionary<string, string>`. However, unlike your standard dictionary, configurations can have named sections to help scope related settings together. Each source of a configuration encodes these sections differently. For example, JSON configurations start a new section with each nested object, whereas environment variables use the double underscore to indicate a new section name (eg. `"first__second__third"`). However, regardless of where they're from, the `IConfiguration` interface surfaces these possibly nested settings in a flattened case-insensitive structured such that the key for a nested value is specified using the colon (`':'`) character to separate section names.

For example, suppose you have the following JSON configuration (taken from the [MSDN logging docs](https://docs.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#configure-logging)):
```json
{
  "Logging": {
    "LogLevel": {
      "Default": "Information",
      "Microsoft": "Warning",
      "Microsoft.Hosting.Lifetime": "Information"
    }
  }
}
```

If you have an `IConfiguration` object for this file, you could access the value of the default log level using the key `"Logging:LogLevel:Default"` which would return `"Information"`. 

So, what is `IConfigurationSection`? Well, `IConfigurationSection` objects are `IConfiguration` objects that have been scoped to a particular section. In fact, they extend the `IConfiguration` interface by providing a `Key`, `Path` (in case we're multiple sections down), and optional `Value`. Practically speaking, an `IConfigurationSection` object operates the same as an `IConfiguration` object except that when you query keys from an `IConfigurationSection`, [it queries the root configuration for its own `Path` + the desired key](https://github.com/dotnet/runtime/blob/3a25a7f1cc446b60678ed25c9d829420d6321eba/src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationSection.cs#L85). Hopefully this example makes things a bit clearer:
```csharp
IConfiguration config = /* the above config */;

// We can fetch the section for LogLevels
IConfigurationSection section = config.GetSection("Logging:LogLevel"); // Or we could say config.GetSection("Logging").GetSection("LogLevel")
Console.WriteLine(section.Key) // "LogLevel"
Console.WriteLine(section.Path) // "Logging:LogLevel"
Console.WriteLine(section.Value) // null
Console.WriteLine(section["Microsoft"]) // "Warning"

// We can also stay up a level
section = config.GetSection("Logging");
Console.WriteLine(section.Key) // "Logging"
Console.WriteLine(section.Path) // "Logging"
Console.WriteLine(section.Value) // null
Console.WriteLine(section["LogLevel:Microsoft"]) // Also "Warning"
```

You can also fetch an `IConfigurationSection` for a key/value pair such that its value is non-`null`:
```csharp
IConfiguration config = /* the above config */;

// We can fetch the section for LogLevels
IConfigurationSection section = config.GetSection("Logging:LogLevel:Microsoft");
Console.WriteLine(section.Key) // "Microsoft"
Console.WriteLine(section.Path) // "Logging:LogLevel:Microsoft"
Console.WriteLine(section.Value) // "Warning"
```

It's by using these sections that we can disambiguate between whether a user is relying on the connection string (which will be a key/value pair in the config) or on an identity (which will be a section name with 1 or more keys); we can check by simply seeing if `IConfigurationSection.Value` is populated.

Now onto the `Get<T>` method. This method creates an object of type `T` and binds the value of each key with the same name as a property of T (with some additional intelligence around parsing strings as integral numbers and other types). For example, imagine a user has the following *local.settings.json*:
```json
{
  "IsEncrypted": false,
  "Values": {
    "AzureWebJobsStorage__blobServiceUri": "https://foo.blob.core.windows.net/",
    "AzureWebJobsStorage__queueServiceUri": "https://foo.queue.core.windows.net/",
    "AzureWebJobsStorage__tableServiceUri": "https://foo.table.core.windows.net/",
  }
}
```

By default, the connection name is `"AzureWebJobsStorage"`, so we'll fetch `IConfigurationSection` whose path is `"AzureWebJobsStorage"`. Now, we'll attempt to fetch its `Value` which should be `null`, so instead we'll bind the section to a value of type `AzureStorageAccountOptions`. The `BlobServiceUri`, `QueueServiceUri`, and `TableServiceUri` properties all have matching keys in the section, so they are bound to corresponding values and parsed as `Uri` objects. There is no `AccountName` key, so its value is left as `null`, giving us a well-typed representation for the config complete with type-checking!

Reply:
I'll also include a sample setup in the description

Reply:
Thanks for this super-detailed explanation! This is much clearer to me now. It seems there are still a lot of new features in .NET Core that I still need to learn. 😅

Reply:
I came from a .NET Framework background too, and it took me a bit to ramp up on all of these new things lol. There's still a lot to digest!